sift.js

```diff
@@ -81,6 +98,350 @@
     return true;
   }
 
+  function smartParseFloat(value) {
+    // Convert strings to numbers where possible so that
+    // strings that represent numbers are compared as numbers
+    if (typeof value !== 'string') {
+      return value;
+    }
+
+    if (isNaN(parseFloat(value.trim()))) {
+      return value;
+    }
+
+    if (parseFloat(value.trim()).toString() !== value.trim()) {
+      return value;
+    }
+
+    return parseFloat(value);
+  }
+
+  function splitByCommas(str, returnNilAsArray) {
+    if (str === undefined || str === null) {
+      return returnNilAsArray === false ? str : [];
+    }
+
+    if (Array.isArray(str)) {
+      return str.map(splitByCommas).flat();
+    }
+
+    if (typeof str !== 'string') {
+      return [str];
+    }
+
+    // Split a string by commas but ignore commas within double-quotes
+    // Turn values within square brackets into a nested array
+    // Adapted from: https://stackoverflow.com/questions/11456850/split-a-string-by-commas-but-ignore-commas-within-double-quotes-using-javascript
+
+    var csvArrayPattern = /(".*?"|\[.*?\]|[^",\[\]]+)(?=\s*,|\s*$)/g;
+    var arrayPattern = /^(?:\[[\w\W]*\])$/;
+    var arr = [];
+    var res = csvArrayPattern.exec(str);
+
+    while (res !== null) {
+      if (arrayPattern.test(res[0])) {
+        arr.push(splitByCommas(res[0].replace(/(?:^\[)|(?:\]$)/g, '').trim()));
+      } else {
+        arr.push(res[0].replace(/(?:^")|(?:"$)/g, '').trim());
+      }
+
+      res = csvArrayPattern.exec(str);
+    }
+
+    return arr.map(function(value) {
+      if (Array.isArray(value)) {
+        return value;
+      }
+
+      return ('' + value).trim();
+    }).filter(function(value) {
+      return Array.isArray(value) || [undefined, null, '', NaN].indexOf(value) === -1;
+    });
+  }
+
+  function getCachedDate(offsetType, offsetValue, getDate) {
```

Comment:
Add JS doc for this function

Reply:
As [this comment](https://weboo.atlassian.net/browse/ID-1493?focusedCommentId=25714) suggested, the PR will remain open. https://weboo.atlassian.net/browse/ID-1802 will require some of these date comparison functions and helper functions to be rewritten. Some of them may become redundant.

Once they're all cleaned up, I'll add the JSDoc for you to review.